api/src/lib.rs

```diff
@@ -1214,6 +1298,73 @@ impl Client {
         Ok(http_response)
     }
 
+    fn splitable_request<LocationT, RequestT, SuccessT, QueryT>(
+        &self,
+        method: Method,
+        url: LocationT,
+        body: RequestT,
+        query: Option<QueryT>,
+        retry: Retry,
+    ) -> Result<SplitableRequestResponse<SuccessT>>
+    where
+        LocationT: IntoUrl + Display + Clone,
+        RequestT: Serialize + SplittableRequest + Clone + Debug,
+        QueryT: Serialize + Clone,
+        for<'de> SuccessT: Deserialize<'de> + ReducableResponse + Clone + Debug,
+    {
+        debug!("Attempting {} `{}`", method, url);
+        let result: Result<SuccessT> = self.request(
+            method.clone(),
+            url.clone(),
+            Some(body.clone()),
+            query.clone(),
+            retry.clone(),
+        );
+
+        match result {
+            Ok(response) => Ok(SplitableRequestResponse {
+                response,
+                num_failed: 0,
+            }),
+            Err(e) => {
+                if Self::should_split(&e) {
+                    let mut num_failed = 0;
+                    let mut response = SuccessT::empty();
+
+                    body.split().iter().for_each(|request| {
+                        match self.request(
+                            method.clone(),
+                            url.clone(),
+                            Some(request),
+                            query.clone(),
+                            retry.clone(),
+                        ) {
+                            Ok(r) => response = SuccessT::merge(response.clone(), r),
```

Comment:
Hmm, this mutation and cloning doesn't look quite right. Conceptually we should be able to [`reduce`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.reduce) the successful responses, although we need to filter out and handle the failures somehow.